### PR TITLE
refactor:  add type safety & fix ci 

### PR DESCRIPTION
**Pull request by KnownBlackHat**

### CODE REVIEW

**Comment by KnownBlackHat** [comment]
This codebase was full of bugs. ig i reduced almost all bugs but some still exists

**Review by shahriyardx — COMMENTED** [review]
Can't we make the cache how much user's actually want? 

```py
_raw_limit = os.getenv("EASY_PIL_CACHE_LIMIT")
cache_limit = int(_raw_limit) if _raw_limit else 32
```

so users can set their own cache limit by defining that environment variable

File: easy_pil/canvas.py [hunk 1]
```diff
@@ -27,15 +27,15 @@ class Canvas:
...
-        width: float = None,
-        height: float = None,
-        color: Color = None,
+        size: Optional[Tuple[int, int]] = None,
+        width: int = 0,
+        height: int = 0,
+        color: Color = 0,
     ) -> None:
-        if not size and not width and not height:
+        if not (size or (width and height)):
```
Comment (shahriyardx) [inline easy_pil/canvas.py:35]: For some reason this actually causes the canvas test to fail
maybe a better approach is 

```py
if not any([size, width, height]):
    raise ValueError("size, width, and height cannot all be None")

if size:
    self.size = size
elif width and height:
    self.size = (width, height)
else:
    raise ValueError("width, and height both must be provided")
```

**Review by shahriyardx — CHANGES_REQUESTED** [review]

File: tests/test_editor.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def test_rotate(self):
     def test_blur(self):
         """Tests editor blur"""
         canvas = Canvas((100, 100), color="black")
-        editor = Editor(canvas).blur(mode="gaussian", amount=10)
+        editor = Editor(canvas).blur(mode="gussian", amount=10)
```
Comment (shahriyardx) [inline tests/test_editor.py:74]: Gaussian is actually the correct spelling [ref](https://en.wikipedia.org/wiki/Gaussian_blur)

**Review by shahriyardx — CHANGES_REQUESTED** [review]

**Comment by KnownBlackHat** [comment]
> Can't we make the cache how much user's actually want?
> 
> ```python
> _raw_limit = os.getenv("EASY_PIL_CACHE_LIMIT")
> cache_limit = int(_raw_limit) if _raw_limit else 32
> ```
> 
> so users can set their own cache limit by defining that environment variable

it will be better if it will be parameterized as there are many use case of PIL things and defining env vars will get quite complex and tricky for users

File: tests/test_editor.py (hunk 2 above)
Comment (KnownBlackHat) [reply tests/test_editor.py:74]: there was  type error as official pil interface gives two mode only `box & gussian`

Comment (KnownBlackHat) [reply tests/test_editor.py:74]: they have restricted literals for above two only

File: easy_pil/canvas.py (hunk 1 above)
Comment (KnownBlackHat) [reply easy_pil/canvas.py:35]: ig this kind of checking will introduce bugs in whole code base

Comment (KnownBlackHat) [reply easy_pil/canvas.py:35]: lemme check the tests again

**Review by shahriyardx — APPROVED** [review]

**Pull request merged**